GitHub.Authentication/Authority.cs

```diff
@@ -50,12 +46,24 @@ internal class Authority : IAuthority
         /// </summary>
         public const int RequestTimeout = 15 * 1000; // 15 second limit
 
-        public Authority(string authorityUrl = null)
+        public Authority(TargetUri targetUri)
         {
-            _authorityUrl = authorityUrl ?? DefaultAuthorityUrl;
+            // The GitHub proper API endpoints
+            if (targetUri.Host == "github.com")
+            {
+                _authorityUrl = "https://api.github.com/authorizations";
+                _validationUrl = "https://api.github.com/user/subscriptions";
+            }
+            else
+            {
+                // If we're here otherwise, it's GitHub Enterprise via configured authority
```

Comment:
Minor nit, but comment is confusing. Should this say

`// If we're here, it's GitHub Enterprise via a configured authority`